File: src/HA2/A3/Mail.java
package src.HA2.A3;

public class Mail {
    private String sender;
    private String senderAddress;
    private String subject;
    private String message;
    private String datetime;
    private boolean read;

    public Mail(String sender, String senderAddress, String subject, String message, String datetime) {
        this.sender=sender;
        this.senderAddress = senderAddress;
        this.subject = subject;
        this.message = message;
        this.datetime = datetime;
        this.read = false;
    }
    public void isRead(){
        this.read =true;
    }

    public String showMail(){
        return subject + " from " + sender + " on " +datetime+ ": " +message;
    }

    public String getSender() {
        return sender;
    }

    public void setSender(String sender) {
        this.sender = sender;
    }

    public String getSenderAddress() {
        return senderAddress;
    }

    public void setSenderAddress(String senderAddress) {
        this.senderAddress = senderAddress;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getDatetime() {
        return datetime;
    }

    public void setDatetime(String datetime) {
        this.datetime = datetime;
    }

    public void setRead(boolean read) {
        this.read = read;
    }

    public boolean getRead(){
        return read;
    }
}

File: python_HAs/HA10/A3/Mail.py
from datetime import datetime

class Mail:
    def __init__(self, subject, sender, message, read):
        self.subject = subject
        self.sender = sender
        self.datetime = datetime.now()
        self.message = message
        self.read =  False

    def mark_as_read(self):
        self.read = True

    def print_full(self):
        formatted_time = self.datetime.strftime("%d.%m.%Y, %H:%M")
        print(f"{self.subject} from {self.sender} on {formatted_time}: {self.message}")
